ui/src/app/pages/homepage/data-access-requirements.tsx

```diff
@@ -0,0 +1,531 @@
+    content={'For additional security, we require you to verify your identity by uploading a photo of your ID.'}>
+  <InfoIcon style={{margin: '0 0.3rem'}}/>
+</TooltipTrigger>;
+
+// TODO merge with RegistrationTasks after we remove RegistrationDashboard
+const moduleLabels: Map<AccessModule, JSX.Element> = new Map([
+  [AccessModule.TWOFACTORAUTH, <div>Turn on Google 2-Step Verification</div>],
+  [AccessModule.RASLINKLOGINGOV, <div>Verify your identity with Login.gov <LoginGovTooltip/></div>],
+  [AccessModule.ERACOMMONS, <div>Connect your eRA Commons account</div>],
+  [AccessModule.COMPLIANCETRAINING, <div>Complete <AoU/> research Registered Tier training</div>],
+  [AccessModule.DATAUSERCODEOFCONDUCT, <div>Sign Data User Code of Conduct</div>],
+]);
+
+// this function does double duty:
+// - returns appropriate text for completed and bypassed modules and null for incomplete modules
+// - because of this, truthy return values indicate that a module is either complete or bypassed
+const bypassedOrCompletedText = (status: AccessModuleStatus) => {
+  const {completionEpochMillis, bypassEpochMillis}: AccessModuleStatus = status || {};
+  const userCompletedModule = !!completionEpochMillis;
+  const userBypassedModule = !!bypassEpochMillis;
+
+  return cond(
+      [userCompletedModule, () => `Completed on: ${displayDateWithoutHours(completionEpochMillis)}`],
+      [userBypassedModule, () => `Bypassed on: ${displayDateWithoutHours(bypassEpochMillis)}`],
+      // return nothing if there's no text
+    () => null
+  );
+};
+
+const handleTerraShibbolethCallback = (token: string, spinnerProps: WithSpinnerOverlayProps) => {
+  const handler = withErrorModal({
+    title: 'Error saving NIH Authentication status.',
+    message: 'An error occurred trying to save your NIH Authentication status. Please try again.',
+    onDismiss: () => {
+      spinnerProps.hideSpinner();
+    }
+  })(async() => {
+    spinnerProps.showSpinner();
+    await profileApi().updateNihToken({jwt: token});
+    spinnerProps.hideSpinner();
+  });
+
+  return handler();
+};
+
+const handleRasCallback = (code: string, spinnerProps: WithSpinnerOverlayProps) => {
+  const handler = withErrorModal({
+    title: 'Error saving RAS Login.Gov linkage status.',
+    message: 'An error occurred trying to save your RAS Login.Gov linkage status. Please try again.',
+    onDismiss: () => {
+      spinnerProps.hideSpinner();
+    }
+  })(async() => {
+    spinnerProps.showSpinner();
+    await profileApi().linkRasAccount({ authCode: code, redirectUrl: buildRasRedirectUrl() });
+    spinnerProps.hideSpinner();
+  });
+
+  return handler();
+};
+
+interface ModuleProps {
+  module: AccessModule;
+  active: boolean;    // is this the currently-active module that the user should complete
+
+  // TODO RW-7059
+  // eligible: boolean;  // is the user eligible to complete this module (does the inst. allow it)
+}
+const MaybeModule = (props: ModuleProps): JSX.Element => {
```

Comment:
nit: might be worth a comment here to describe under what conditions this would yield null